### PR TITLE
Water atcor develop

### PR DESCRIPTION
**Pull request by passangd**
sun-glint correction methods and modification of workflow to enable aerosol input from users

### CODE REVIEW

File: wagl/reflectance.py [hunk 1]
```diff
@@ -786,16 +902,11 @@ def calculate_reflectance(
...
-        # NOTES:
-        #    for the time being we're outputting float32 while water atcor details
-        #    are figured out on how sun and sky glint is to be applied.
-        #    it is unknown that if we truncate to int16 now, will this have an
-        #    impact applying glint correction later
-        kwargs["fillvalue"] = numpy.nan
-        kwargs["dtype"] = "float32"
-
-        # update the attrs for the float32 datasets
-        attrs["no_data_value"] = numpy.nan
```
Comment (sixy6e) [inline wagl/reflectance.py:798]: We still need this attrs item. Yes it is duplication, but for the time being there is still functions depending on it. Sometime in the future we'll simply just use the `fillvalue` item.

**Review by sixy6e — APPROVED** [review]
Looks good.
In regards to the comment regarding the attrs item `no_data_value`, in future, we'll migrate over to simply using `fillvalue`, so just keep a mental note of this.

File: wagl/reflectance.py (hunk 1 above)
Comment (passangd) [reply wagl/reflectance.py:798]: attrs["no_data_value"] is same  as previously defined on line 864. So this is simply duplication unless our no_data_value changed.

**Review by sixy6e — APPROVED** [review]
good to merge

**Pull request merged**